core/primitives/transaction_validity.hpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_PRIMITIVES_TRANSACTION_VALIDITY_HPP_
+#define KAGOME_CORE_PRIMITIVES_TRANSACTION_VALIDITY_HPP_
+
+#include <variant>
+#include <vector>
+
+namespace kagome::primitives {
+
+  namespace {
```

Comment:
What is the purpose of anonymous namespace here? It's generally a very bad idea to use them in headers:
https://rules.sonarsource.com/cpp/RSPEC-1000
https://stackoverflow.com/questions/357564/uses-for-anonymous-namespaces-in-header-files

Reply:
If you want to hide a definition of structure `Valid`, you can forward declare it here and expand in .cpp

Reply:
We need a variant<...>, so hiding it in cpp will not work. I don't like anonymous namespace in hpp though...